app/src/test/java/com/example/tictactoe/model/TicTacToeViewModelTest.kt

```diff
@@ -0,0 +1,162 @@
+import com.example.tictactoe.R
+import com.example.tictactoe.data.source.FakeGameBoardPart
+import com.example.tictactoe.util.getOrAwaitValue
+import org.hamcrest.CoreMatchers.`is`
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+class TicTacToeViewModelTest {
+
+    @get:Rule // Ensures liveData doesn't run on main thread
+    var instantTaskExecutorRule = InstantTaskExecutorRule()
+
+
+    private lateinit var viewModel: TicTacToeViewModel
+
+    @Before
+    fun setup_view_model() {
+        viewModel = TicTacToeViewModel()
+    }
+
+    @After
+    fun clean_up_view_model_and_game_board() {
+        viewModel.resetGameState()
+        FakeGameBoardModel.resetState()
+    }
+
+    @Test
+    fun set_top_horizontal_row_test_row_returns_true() {
+
+        FakeGameBoardModel.testWin(0, FakeGameBoardModel.Horizontal())
+
+        assertThat("Testing win did not return true for top row",
+            viewModel.testRow(FakeGameBoardPart.ROW_TOP.part, FakeGameBoardPart.BOX_TOP_LEFT.part),
+            `is`(true)
+        )
+    }
+
+    @Test
+    fun set_middle_horizontal_row_test_row_returns_true() {
+
+        FakeGameBoardModel.testWin(1, FakeGameBoardModel.Horizontal())
+
+        assertThat("Testing win did not return true for middle row",
+            viewModel.testRow(FakeGameBoardPart.ROW_MIDDLE.part, FakeGameBoardPart.BOX_CENTER.part),
+            `is`(true)
+        )
+    }
+
+    @Test
+    fun set_bottom_horizontal_row_test_row_returns_true() {
+
+        FakeGameBoardModel.testWin(2, FakeGameBoardModel.Horizontal())
+
+        assertThat("Testing win did not return true for bottom row",
+            viewModel.testRow(FakeGameBoardPart.ROW_BOTTOM.part, FakeGameBoardPart.BOX_BOTTOM_RIGHT.part),
+            `is`(true)
+        )
+    }
+
+    @Test
+    fun set_left_vertical_column_test_column_returns_true() {
+
+        FakeGameBoardModel.testWin(0, FakeGameBoardModel.Vertical())
+
+        assertThat("Testing win did not return true for left column",
+            viewModel.testRow(FakeGameBoardPart.COLUMN_LEFT.part, FakeGameBoardPart.BOX_TOP_LEFT.part),
+            `is`(true)
+        )
+    }
+
+    @Test
+    fun set_middle_vertical_column_test_column_returns_true() {
+
+        FakeGameBoardModel.testWin(1, FakeGameBoardModel.Vertical())
+
+        assertThat("Testing win did not return true for middle column",
+            viewModel.testRow(FakeGameBoardPart.COLUMN_MIDDLE.part, FakeGameBoardPart.BOX_CENTER.part),
+            `is`(true)
+        )
+    }
+
+    @Test
+    fun set_right_vertical_column_test_column_returns_true() {
+
+        FakeGameBoardModel.testWin(2, FakeGameBoardModel.Vertical())
+
+        assertThat("Testing win did not return true for right column",
+            viewModel.testRow(FakeGameBoardPart.COLUMN_RIGHT.part, FakeGameBoardPart.BOX_BOTTOM_RIGHT.part),
+            `is`(true)
+        )
+    }
+
+    @Test
```

Comment:
For some reason when I run all the tests together this one will pass (along with the last two) but the rest fail.

Independently they all pass.

@j-roskopf Do you have any clue what I might be missing?